### PR TITLE
[c/en] Just a few explanations, a couple of tiny changes

### PR DESCRIPTION
**Pull request by Naereen**
Just add a few explanations where I thought details were missing:

- "Floating-point numbers and calculations are not exact" was mentioned without any examples. Many students of mine believe that floating-point calculation errors are very rare and esoteric, but they are not!
- Just a little bit of more details for "goto" and labels.

- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)

### CODE REVIEW

File: c.html.markdown [hunk 1]
```diff
@@ -349,26 +355,28 @@ int main (int argc, char** argv)
...
-  error :
+  error: // this is a label that you can "jump" to with "goto error;"
   printf("Error occurred at i = %d & j = %d.\n", i, j);
   /*
-  https://ideone.com/GuPhd6
-  this will print out "Error occurred at i = 51 & j = 99."
+    https://ideone.com/GuPhd6
+    this will print out "Error occurred at i = 51 & j = 99."
   */
+  /* it is generally considered bad practice to do so, except if  */
```
Comment (Menelion) [inline c.html.markdown:378]: Join those two comments and make it one single comment or convert both to single-line `//`-prepended comments?

File: c.html.markdown [hunk 2]
```diff
@@ -741,6 +749,7 @@ typedef void (*my_fnp_type)(char *);
 // Order of Evaluation
 ///////////////////////////////////////
 
+// From top to bottom, top is has higher precedence
```
Comment (Menelion) [inline c.html.markdown:752]: A mistake here?

**Review by Menelion — CHANGES_REQUESTED** [review]

File: c.html.markdown (hunk 2 above)
Comment (Naereen) [reply c.html.markdown:752]: Oh you're right, "top has". I'll update.

File: c.html.markdown [hunk 3]
```diff
@@ -224,9 +224,15 @@ int main (int argc, char** argv)
...
+  
   // Floating-point numbers and calculations are not exact
+  // for instance it is not giving mathematically correct results
+  (0.1 + 0.1 + 0.1) != 0.3; // => 1 (true)
+  // and it is NOT associative
+  1 + (1e123 - 1e123) != (1 + 1e123) - 1e123; // => 1 (true)
+  // this notation is scientific notations for numbers: 1e123 = 1*10^123
 
-  // Modulo is there as well
+  // Modulo is there as well, but be careful if arguments are negative
```
[the remark this answers is not in the document]
Comment (Naereen) [reply c.html.markdown:235]: Well the issue is that I'm not a C expert and not sure what happens with negative arguments.
I know that in Python or OCaml, it's advised not to trust the results.
I'll write a small C program and see what happens.

[the remark this answers is not in the document]
Comment (Naereen) [reply c.html.markdown:235]: Yep, it's as I thought, a negative sign on first or second argument does not yield the same result:
```
10 % 2 = 0
10 % 3 = 1
10 % 4 = 2
10 % 5 = 0
-10 % 2 = 0
-10 % 3 = -1  // ok, makes sens
-10 % 4 = -2  // ok, makes sens
-10 % 5 = 0
10 % -2 = 0
10 % -3 = 1  // where is the sign?
10 % -4 = 2  // where is the sign?
10 % -5 = 0
How weird is that?
```
(see attached C file)
[test.txt](https://github.com/adambard/learnxinyminutes-docs/files/5899716/test.txt)

**Comment by Naereen** [comment]
@Menelion thanks for the reviews, I followed your suggestions and updated my patch and PR.
I think you can now merge safely. Thanks!

**Comment by Naereen** [comment]
Hi @adambard, could you merge? Thanks :sun_with_face: !

**Comment by adambard** [comment]
LGTM, thanks for your contribution!

**Pull request merged**